File: LeetCode/Python/0075. Sort Colors #1.py
class Solution:
    def sortColors(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        def swap(arr, left, right):
            arr[left], arr[right] = arr[right], arr[left]
        
        j = 0
        for i in range(len(nums)):
            if nums[i] == 0:
                swap(nums, i, j)
                j += 1
        
        j = len(nums) - 1
        for i in range(len(nums) - 1, -1, -1):
            if nums[i] == 2:
                swap(nums, i, j)
                j -= 1
        

File: LeetCode/Java/0075. Sort Colors #1.java
class Solution {
    public void sortColors(int[] nums) {
        int n = nums.length;
        
        int j = 0;
        for (int i = 0; i < n; i++)
            if (nums[i] == 0)
                swap(nums, i, j++);
        
        j = n - 1;
        for (int i = n - 1; i >= 0; i--)
            if (nums[i] == 2)
                swap(nums, i, j--);
    }
    
    private void swap(int[] arr, int left, int right) {
        int temp = arr[left];
        arr[left] = arr[right];
        arr[right] = temp;
    }
}
